reposerver/repository/lock.go

```diff
@@ -43,27 +43,24 @@ func (r *repositoryLock) Lock(path string, revision string, allowConcurrent bool
 
 	for {
 		state.cond.L.Lock()
+		defer state.cond.L.Unlock()
```

Comment:
defer will not work in for loop in this case and will cause leak. Also tests catch this issue